virtcontainers/sandbox.go

```diff
@@ -1166,6 +1167,11 @@ func (s *Sandbox) CreateContainer(contConfig ContainerConfig) (VCContainer, erro
 	defer func() {
 		// Rollback if error happens.
 		if err != nil {
+			s.Logger().Warningf("Container %q could not be created, stopping it", contConfig.ID)
+			if err = c.stop(true); err != nil {
+				s.Logger().WithError(err).WithField("container-id", c.id).WithField("sandboxid", s.id).Warning("Could not delete container")
```

Comment:
Thanks for adding the structured logging. You can do this more consistently and simply though by creating a new logger. Something like:

```diff
diff --git a/virtcontainers/sandbox.go b/virtcontainers/sandbox.go
index a4e5327a..dd42104d 100644
--- a/virtcontainers/sandbox.go
+++ b/virtcontainers/sandbox.go
@@ -1167,11 +1167,16 @@ func (s *Sandbox) CreateContainer(contConfig ContainerConfig) (VCContainer, erro
        defer func() {
                // Rollback if error happens.
                if err != nil {
-                       s.Logger().Warningf("Container %q could not be created, stopping it", contConfig.ID)
-                       if err = c.stop(true); err != nil {
-                               s.Logger().WithError(err).WithField("container-id", c.id).WithField("sandboxid", s.id).Warning("Could not delete container")
+                       logger := s.Logger().WithFields(logrus.Fields{"container-id": c.id, "sandox-id": s.id, "rollback": true})
+
+                       logger.Warning("Cleaning up partially created container")
+
+                       if err2 := c.stop(true); err2 != nil {
+                               logger.WithError(err2).Warning("Could not delete container")
                        }
-                       s.Logger().WithField("container-id", c.id).WithField("sandboxid", s.id).Info("Container was stopped. Removing from sandbox store")
+
+                       logger.Debug("Removing stopped container from sandbox store")
+
                        s.removeContainer(c.id)
                }
        }()
```

(Note too the "rollback" log field for clarity)